include/eve/module/core/regular/impl/is_less.hpp

```diff
@@ -0,0 +1,42 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+  template<typename T, typename U, callable_options O>
+  EVE_FORCEINLINE constexpr common_logical_t<T, U> is_less_(EVE_REQUIRES(cpu_), O const& o, T a, U b) noexcept
+  {
+    if constexpr (O::contains(definitely))
+    {
+      using w_t = common_value_t<T, U>;
+      auto aa = w_t{a};
+      auto bb = w_t{b};
+
+      auto tol = o[definitely].value(w_t{});
+
+      if constexpr (integral_value<decltype(tol)>) return aa < eve::prev(bb, tol);
+      else                                         return aa < fam(bb, -tol, eve::max(eve::abs(aa), eve::abs(bb)));
+    }
+    else
+    {
+      if constexpr (scalar_value<T>)
+      {
+        if      constexpr (scalar_value<U>)                    return common_logical_t<T, U>(a < b);
+        else if constexpr (std::same_as<T, element_type_t<U>>) return is_less(U{a}, b);
+        else                                                   return is_less(element_type_t<U>{a}, b);
+      }
+      else
+      {
+        if      constexpr (simd_value<U>)                      return map(is_less, a, b);
+        else if constexpr (std::same_as<element_type_t<T>, U>) return is_less(a, T{b});
+        else                                                   return is_less(a, element_type_t<T>{b});
+      }
+    }
+  }
```

Comment:
Just for me - where is the mask handled in generaic case?
I see handling in avx512 but that's it.

Reply:
It is handled in the callable

Reply:
Like always false? Or what does it mean?

Reply:
noone answered me

Reply:
The general masking (ie not using AVX512/SVE masking) is done in elementwise_callable that detect a non trivial mask, etxract it and call perform_mask.

See : https://github.com/jfalcou/eve/blob/71d23596c5d5ba1f008f1f729388d952aa46d6ce/include/eve/traits/overload/default_behaviors.hpp#L131

This has been the case since ages.

Reply:
```
          return detail::mask_op(cond, detail::return_2nd, x0, f(x0,xs...));
```

Yeah - ok - this doesn't seem right for is_less. I don't even understand how it works.

Reply:
Good question I will check 
I am a bit wary about masking predicates anyway